### DISCUSSION TITLE
Generate rust types from node-types.json

### DISCUSSION THREAD

**Discussion by JonasCir** [tree-sitter/tree-sitter#1803] [Libraries & tools] [answered]
Hi there,
the official documentation mentions [here](https://tree-sitter.github.io/tree-sitter/using-parsers#static-node-types) that `node-types.json` can be used "to generate type declarations in statically-typed programming languages". This is exactly what I need :smiley: .

Is there any tool which creates for a given `node-types.json` file the corresponding Rust structs/types? 

If not, this would be a project I would definitely create and contribute to.

Best regards,
Jonas

**Comment by feathecutie** [comment, marked as answer]
In case this is still relevant for you, or if anyone else stumbles upon this question looking for the same thing:
There is a [`type-sitter` crate on `crates.io`](https://crates.io/crates/type-sitter) which seems to do precisely this.
